lib/awful/completion.lua

```diff
@@ -90,7 +90,7 @@ function completion.shell(command, cur_pos, ncomp, shell)
         i = i + 1
     end
 
-    if cword_index == 1 and not string.find(words[cword_index], "/") then
+    if cword_index == 1 then
```

Comment:
If I remember correctly it is a hack: if there's no '/' then it's considered a command, but that fails for `/usr/bin/ls` etc, of course.